Tools/Scripts/webkitpy/w3c/common.py

```
@@ -58,13 +58,12 @@ def read_credentials(host, credentials_json):
 
 
 class WPTPaths:
-    CHECKOUT_DIRECTORY = ["w3c-tests"]
-    WPT_CHECKOUT_PATH = CHECKOUT_DIRECTORY + ["web-platform-tests"]
+    WPT_CHECKOUT_PATH = "web-platform-tests"
 
     @staticmethod
     def checkout_directory(finder):
-        return finder.path_from_webkit_outputdir(*WPTPaths.CHECKOUT_DIRECTORY)
+        return os.path.join(os.path.dirname(finder.webkit_base()))
 
     @staticmethod
     def wpt_checkout_path(finder):
-        return finder.path_from_webkit_outputdir(*WPTPaths.WPT_CHECKOUT_PATH)
+        return os.path.join(os.path.dirname(finder.webkit_base()), WPTPaths.WPT_CHECKOUT_PATH)
```

Is there any benefit to having this, when it isn't necessarily accurate?

I realise you're still using this to get the default path, but it feels potentially misleading that it doesn't return the "current" path.

My gut would actually be to make this function own the logic you put in `test_exporter.py`; that means changing the signature to something like `wpt_checkout_path(finder: WebKitFinder, *, non_interactive: bool = True) -> str` (and I'd probably just make it raise if non-interactive, at least until we're reading from some config).

We probably also just want to wrap it with `@memotize` to avoid prompting multiple times!

---

Thanks for the feedback! Here are the main changes in the latest update, most of which we've discussed offline already:
- Rename `wpt_checkout_path` to `default_wpt_checkout_path`.
- Create `WPTPaths.ensure_wpt_repository` which now owns the logic and cloning from `test_exporter.py`.
    - This is now also used in `test_importer.py`, and enables cloning into a user given directory.
- If in non-interactive mode, we just clone to the default location. I believe this makes more sense than failing the entire operation.